clang/lib/AST/Type.cpp

```diff
@@ -2669,6 +2669,8 @@ bool QualType::isTriviallyRelocatableType(const ASTContext &Context) const {
     return false;
   } else if (const auto *RD = BaseElementType->getAsRecordDecl()) {
     return RD->canPassInRegisters();
+  } else if (BaseElementType.isTriviallyCopyableType(Context)) {
+    return true;
   } else {
     switch (isNonTrivialToPrimitiveDestructiveMove()) {
```

Comment:
I'm not sure why these are different functions, other than maybe having different preconditions.

Reply:
You mean why is `isNonTrivialToPrimitiveDestructiveMove` different from `isTriviallyRelocatable`? I don't know, but I think it has to do with Objective-C, so I left it alone.